packages/gradle/src/utils/version-catalog-utils.ts

```diff
@@ -0,0 +1,189 @@
+import { Tree, globAsync } from '@nx/devkit';
+import * as TOML from 'smol-toml';
+import { gradleProjectGraphPluginName } from './versions';
+
+export async function findVersionCatalogFiles(tree: Tree): Promise<string[]> {
+  const versionCatalogPaths: string[] = [];
+
+  tree.listChanges().forEach((change) => {
```

Comment:
Do you know why we need to look through the changes in the tree when we're already doing the glob?

The glob below should also pick up files which were written to the tree but not on disk yet.

cc @AgentEnder

Reply:
Woop. I think you-know-what dropped that there. Yeah the glob itself is enough.